### PR TITLE
Source Code and Artifact Directory Restructure

### PR DESCRIPTION
**Pull request by ohsh6o**
As we have a collection of source code for OSCAL content and validation tools, begin better directory structure and organization to clarify what is source material & code versus what are resulting built artifacts.

Done in coordination with the 10x ASAP Team for 18F/fedramp-automation#131.

### CODE REVIEW

File: src/web/package.json [hunk 1]
```diff
@@ -8,10 +8,10 @@
...
-    "build:sch-stage-1": "xslt3 -s:../src/ssp.sch -xsl:../lib/schematron/trunk/schematron/code/iso_dsdl_include.xsl -o:validations/stage1.sch",
-    "build:sch-stage-2": "xslt3 -s:validations/stage1.sch -xsl:../lib/schematron/trunk/schematron/code/iso_abstract_expand.xsl -o:validations/stage2.sch",
-    "build:sch-stage-3": "xslt3 -s:validations/stage2.sch -xsl:../lib/schematron/trunk/schematron/code/iso_svrl_for_xslt2.xsl -o:validations/ssp.xsl",
+    "build:schematron-java": "cd ../validations && ./bin/validate_with_schematron.sh && cp target/ssp.xsl ../web/public",
+    "build:sch-stage-1": "xslt3 -s:../validations/rules/ssp.sch -xsl:../../vendor/schematron/trunk/schematron/code/iso_dsdl_include.xsl -o:../validations/target/ssp-stage1.sch",
+    "build:sch-stage-2": "xslt3 -s:../validations/target/ssp-stage1.sch -xsl:../../vendor/schematron/trunk/schematron/code/iso_abstract_expand.xsl -o:../validations/target/ssp-stage2.sch",
+    "build:sch-stage-3": "xslt3 -s:../validations/target/ssp-stage2.sch -xsl:../../vendor/schematron/trunk/schematron/code/iso_svrl_for_xslt2.xsl -o:../validations/target/ssp.xsl",
```
Comment (ohsh6o) [inline src/web/package.json:14]: @danielnaab, I have come back to this gem but cannot figure out why.

```sh
> npx xslt3 -s:./stage2.sch -xsl:../lib/schematron/trunk/schematron/code/iso_svrl_for_xslt2.xsl -o:./stage3.sch
Transformation failure: Error SESU0013
    Serializer does not support the requested XML version: 2.0
Error SESU0013
    Serializer does not support the requested XML version: 2.0
```

We discussed verbally but I didn't record [in our Slack thread](https://flexion.slack.com/archives/C020WVDQFPE/p1622127533004000) or elsewhere how to move beyond it. Ideas?

File: src/web/package.json [hunk 2]
```diff
@@ -8,10 +8,10 @@
     "build": "snowpack build",
```
Comment (ohsh6o) [inline src/web/package.json:8]: @danielnaab: also getting `esbuild` weirdness and I have not changed much so I am confuzzled, now it breaks Federalist builds. :-(

```sh
$ pwd
/Users/myusername/Code/fedramp-automation
$ npm run federalist
$ npm run federalist

> fedramp-automation@0.0.1 federalist
> npm run build:validation-ui && npm run link:validation-ui


> fedramp-automation@0.0.1 build:validation-ui
> cd src/web && npm install && npm run build && cd ../../..


> postinstall
> patch-package

patch-package 6.4.7
Applying patches...
overmind-react@28.0.0 ✔

up to date, audited 1725 packages in 3s

9 moderate severity vulnerabilities

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

> build
> snowpack build

[19:23:21] [snowpack] ! building files...
/Users/myusername/Code/fedramp-automation/src/web/node_modules/esbuild/bin/esbuild:2
throw new Error(`esbuild: Failed to install correctly
^

Error: esbuild: Failed to install correctly

Make sure you don't have "ignore-scripts" set to true. You can check this with
"npm config get ignore-scripts". If that returns true you can reset it back to
false using "npm config set ignore-scripts false" and then reinstall esbuild.

If you're using npm v7, make sure your package-lock.json file contains either
"lockfileVersion": 1 or the code "hasInstallScript": true. If it doesn't have
either of those, then it is likely the case that a known bug in npm v7 has
corrupted your package-lock.json file. Regenerating your package-lock.json file
should fix this issue.

    at Object.<anonymous> (/Users/myusername/Code/fedramp-automation/src/web/node_modules/esbuild/bin/esbuild:2:7)
    at Module._compile (node:internal/modules/cjs/loader:1109:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1138:10)
    at Module.load (node:internal/modules/cjs/loader:989:32)
    at Function.Module._load (node:internal/modules/cjs/loader:829:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
    at node:internal/main/run_main_module:17:47
[19:23:21] [snowpack] Build Result Error: There was a problem with a file build result.
[19:23:21] [snowpack] The service was stopped
[19:23:21] [snowpack] Error: The service was stopped
    at /Users/myusername/Code/fedramp-automation/src/web/node_modules/esbuild/lib/main.js:1066:33
    at /Users/myusername/Code/fedramp-automation/src/web/node_modules/esbuild/lib/main.js:568:9
    at Socket.afterClose (/Users/myusername/Code/fedramp-automation/src/web/node_modules/esbuild/lib/main.js:546:7)
    at Socket.emit (node:events:377:35)
    at endReadableNT (node:internal/streams/readable:1312:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
myhostname:fedramp-automation myusername$
```

Comment (ohsh6o) [reply src/web/package.json:8]: Following recommendations from similar issue (evanw/esbuild#462 comments) did not help either.

```sh
$ node node_modules/esbuild/install.js
$ npm run build

> build
> snowpack build

[19:34:13] [snowpack] ! building files...
[19:34:15] [@snowpack/plugin-typescript] Command completed.
[19:34:18] [snowpack] Build Result Error: There was a problem with a file build result.
[19:34:18] [snowpack] Cannot find module 'postcss'
Require stack:
- /Users/myusername/Code/fedramp-automation/src/web/node_modules/@snowpack/plugin-postcss/worker.js
[19:34:18] [snowpack] Error: Cannot find module 'postcss'
Require stack:
- /Users/myusername/Code/fedramp-automation/src/web/node_modules/@snowpack/plugin-postcss/worker.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:941:15)
    at Function.Module._load (node:internal/modules/cjs/loader:774:27)
    at Module.require (node:internal/modules/cjs/loader:1013:19)
    at require (node:internal/modules/cjs/helpers:93:18)
    at Object.<anonymous> (/Users/myusername/Code/fedramp-automation/src/web/node_modules/@snowpack/plugin-postcss/worker.js:4:17)
    at Module._compile (node:internal/modules/cjs/loader:1109:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1138:10)
    at Module.load (node:internal/modules/cjs/loader:989:32)
    at Function.Module._load (node:internal/modules/cjs/loader:829:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
```

Comment (ohsh6o) [reply src/web/package.json:8]: Had to cherry-pick [`package-lock.json`](https://github.com/GSA/fedramp-automation/pull/137/commits/b5fac2794640c1f8bf729b560ae8147fde4fb410) from downstream develop branch to get it to work.

**Review by danielnaab — APPROVED** [review]
Minus whatever you need to do on the Github Actions, looks great. :shipit:

**Pull request merged**